solver/src/main.rs

```diff
@@ -208,38 +209,54 @@ async fn main() {
             .await
             .unwrap();
 
-    let pair_providers =
-        pool_aggregating::pair_providers(&args.shared.baseline_sources, chain_id, &web3).await;
-    let pool_aggregator = PoolAggregator::from_providers(&pair_providers, &web3).await;
-    let pool_fetcher = Arc::new(
-        PoolCache::new(
-            CacheConfig {
-                number_of_blocks_to_cache: args.shared.pool_cache_blocks,
-                // 0 because we don't make use of the auto update functionality as we always fetch
-                // for specific blocks
-                number_of_entries_to_auto_update: 0,
-                maximum_recent_block_age: args.shared.pool_cache_maximum_recent_block_age,
-                max_retries: args.shared.pool_cache_maximum_retries,
-                delay_between_retries: args.shared.pool_cache_delay_between_retries_seconds,
-            },
-            Box::new(pool_aggregator),
-            current_block_stream.clone(),
-            metrics.clone(),
-        )
-        .expect("failed to create pool cache"),
-    );
+    let cache_config = CacheConfig {
+        number_of_blocks_to_cache: args.shared.pool_cache_blocks,
+        // 0 because we don't make use of the auto update functionality as we always fetch
+        // for specific blocks
+        number_of_entries_to_auto_update: 0,
+        maximum_recent_block_age: args.shared.pool_cache_maximum_recent_block_age,
+        max_retries: args.shared.pool_cache_maximum_retries,
+        delay_between_retries: args.shared.pool_cache_delay_between_retries_seconds,
+    };
+    let pool_caches: HashMap<BaselineSources, Arc<PoolCache>> =
+        stream::iter(args.shared.baseline_sources)
```

Comment:
This also seems like something that could be contained in some sort of builder method since it is essentially a map from `BaselineSources -> Arc<PoolCache>` maybe something like 

```rust
impl BaselineSource {
    pub fn to_pool_cache_tuple(&self) -> (Self, Arc<PoolCache>)
}
```

This way we could baseline_sources.iter().map(to_pool_cache_tuple)` and collect into a hashmap...

Just a suggestion to keep main from being so complicated.